### PR TITLE
:sparkles: 210 handling of objects in select component

### PR DESCRIPTION
**Pull request by lehju**
**Description**

Add the handling of lists of objects to select component.

**Reference**

Issues #210

### CODE REVIEW

**Comment by lehju** [comment]
In the storybook, the value of modelValue is always converted into a string. An object cannot be set.
In the Object Select example, the object is set correctly via the modelValue.

**Comment by FabianWilms** [comment]
> In the storybook, the value of modelValue is always converted into a string. An object cannot be set. In the Object Select example, the object is set correctly via the modelValue.

In https://github.com/it-at-m/muc-patternlab-vue/pull/215/files#diff-364ecfb18d69f93eceacc529fb8428c89deaf57b5e56977dcf61829933201e69R19-R26 the default story is defined, which is the one displayed above all other stories.

In this default story a string is used for modelValue and for the selectable items. This is why there is a string displayed.

If you want to show an object value for modelValue you have to convert the default story to a sample with objects.

**Review by FabianWilms — CHANGES_REQUESTED** [review]

File: src/components/Form/MucSelect.vue
Comment (langehm) [reply src/components/Form/MucSelect.vue:140]: or `itemLabel`?

Comment (lehju) [reply src/components/Form/MucSelect.vue]: I have renamed it to ```itemTitle```, as that is what it is called in the vuetify.

File: src/components/Form/MucSelect.vue
Comment (lehju) [reply src/components/Form/MucSelect.vue]: `itemTitle` is now used for the comparison.

File: src/components/Form/MucSelectItem.vue
Comment (lehju) [reply src/components/Form/MucSelectItem.vue]: Done.

File: src/components/Form/MucSelect.vue
Comment (lehju) [reply src/components/Form/MucSelect.vue]: Done.

File: src/components/Form/MucSelect.vue
Comment (lehju) [reply src/components/Form/MucSelect.vue]: Done.

File: src/components/Form/MucSelect.vue
Comment (lehju) [reply src/components/Form/MucSelect.vue]: Done.

File: src/components/Form/MucSelect.vue
Comment (lehju) [reply src/components/Form/MucSelect.vue]: Done.

File: src/components/Form/MucSelect.vue
Comment (lehju) [reply src/components/Form/MucSelect.vue]: Unfortunately I have to make this query, otherwise I get an error.
`Vue: Type any[] | ItemAsObject must have a [Symbol.iterator]() method that returns an iterator.`
Do you have a solution for this?

**Review by FabianWilms — DISMISSED** [review]
Thanks for implementing the feedback - I would be happy to merge this. @langehm waiting for your approval, too.

File: src/components/Form/MucSelectItem.vue
Comment (langehm) [inline src/components/Form/MucSelectItem.vue:13]: I think you forgot one here :)

Comment (langehm) [inline src/components/Form/MucSelectItem.vue:13]:
```suggestion
  item: MucSelectItemTypes;
```
I think you forgot one here + import :)

**Review by langehm — CHANGES_REQUESTED** [review]
Otherwise it lgtm :D

**Review by langehm — APPROVED** [review]

**Pull request merged**
